pkg/plugin/driver/base.go

```diff
@@ -0,0 +1,144 @@
+package driver
+
+import (
+	"crypto/ecdsa"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+
+	"github.com/heptio/sonobuoy/pkg/plugin"
+	"github.com/heptio/sonobuoy/pkg/plugin/manifest"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kuberuntime "k8s.io/apimachinery/pkg/runtime"
+)
+
+// Base is the  truct that stores state for plugin drivers and contains helper methods.
+type Base struct {
+	Definition    plugin.Definition
+	SessionID     string
+	Namespace     string
+	SonobuoyImage string
+	CleanedUp     bool
+}
+
+// TemplateData is all the fields available to plugin driver templates.
+type TemplateData struct {
+	PluginName        string
+	ResultType        string
+	SessionID         string
+	Namespace         string
+	SonobuoyImage     string
+	ProducerContainer string
+	MasterAddress     string
+	CACert            string
+	SecretName        string
+}
+
+// GetSessionID returns the session id associated with the plugin.
+func (b *Base) GetSessionID() string {
```

Comment:
Might be good to mention these getters provide the methods required to satisfy the plugin.Interface